bundles/org.openhab.binding.playstation/src/main/java/org/openhab/binding/playstation/internal/PS4Handler.java

```diff
@@ -0,0 +1,845 @@
+                    readBuffer.position(0);
+                    PS4Command lastCommand = parseResponsePacket(messBuffer);
+
+                    if (lastCommand == PS4Command.SERVER_STATUS_RSP) {
+                        if (oskOpen && isLinked(CHANNEL_OSK_TEXT)) {
+                            sendOSKStart();
+                        } else {
+                            sendStatus();
+                        }
+                    }
+                }
+            } catch (IOException e) {
+                logger.info("Connection read exception: {}", e.getMessage());
+            } finally {
+                try {
+                    channel.close();
+                } catch (IOException e) {
+                    logger.info("Connection close exception: {}", e.getMessage());
+                }
+            }
+            logger.debug("SocketHandler done.");
+            ps4Crypto.clearCiphers();
+            loggedIn = false;
+        }
+    }
+
+    private @Nullable PS4Command parseResponsePacket(ByteBuffer rBuffer) {
+        rBuffer.rewind();
+        final int buffSize = rBuffer.remaining();
+        final int size = rBuffer.getInt();
+        if (size > buffSize || size < 12) {
+            logger.info("Response size ({}) not good, buffer size ({}).", size, buffSize);
+            return null;
+        }
+        int cmdValue = rBuffer.getInt();
+        int statValue = rBuffer.getInt();
+        PS4ErrorStatus status = PS4ErrorStatus.valueOfTag(statValue);
+        PS4Command command = PS4Command.valueOfTag(cmdValue);
+        byte[] respBuff = new byte[size];
+        rBuffer.rewind();
+        rBuffer.get(respBuff);
+        if (command != null) {
+            if (status == null) {
+                logger.debug("Resp; size:{}, command:{}, statValue:{}, data:{}.", size, command, statValue, respBuff);
+            } else {
+                logger.debug("Resp; size:{}, command:{}, status:{}, data:{}.", size, command, status, respBuff);
+            }
+            switch (command) {
+                case LOGIN_RSP:
+                    if (status == null) {
+                        logger.info("Unhandled Login status value: {}", statValue);
+                        return command;
+                    }
+                    // Read login response
+                    switch (status) {
+                        case STATUS_OK:
+                            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, status.message);
+                            loggedIn = true;
+                            if (isLinked(CHANNEL_2ND_SCREEN)) {
+                                scheduler.schedule(() -> {
+                                    ByteBuffer outPacket = PS4PacketHandler
+                                            .makeClientIDPacket("com.playstation.mobile2ndscreen", "18.9.3");
+                                    sendPacketEncrypted(outPacket, false);
+                                }, 10, TimeUnit.MILLISECONDS);
```

Comment:
10ms can be executed in the same time slice, as it is too short for the JVM. If you need a minimal delay, 100ms would be better.